remote-config/src/test/kotlin/com/rakuten/tech/mobile/remoteconfig/SdkHeadersInterceptorSpec.kt

```diff
@@ -0,0 +1,78 @@
+package com.rakuten.tech.mobile.remoteconfig
+
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.amshove.kluent.shouldEqual
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.ParameterizedRobolectricTestRunner
+import java.util.logging.Level
+import java.util.logging.LogManager
+
+@RunWith(ParameterizedRobolectricTestRunner::class)
+class SdkHeadersInterceptorSpec(
+    private val description: String,
+    private val name: String,
+    private val value: String
+) : RobolectricBaseSpec() {
+
+    private val server = MockWebServer()
+
+    init {
+        LogManager.getLogManager()
+            .getLogger(MockWebServer::class.java.name).level = Level.OFF
+    }
+
+    @Before
+    fun setup() {
+        server.start()
+        server.enqueue(MockResponse())
+    }
+
+    @After
+    fun tearDown() {
+        server.shutdown()
+    }
+
+    @Test
+    fun `should attach the header`() {
+        val client = OkHttpClient.Builder().addNetworkInterceptor(
+            SdkHeadersInterceptor(
+                appId = "test-app-id",
+                appName = "test.app.name",
+                appVersion = "1.0.0",
+                deviceModel = "test model name",
+                deviceOsVersion = "2.0.0",
+                sdkVersion = "3.0.0"
+            )
+        ).build()
+
+        client.newCall(
+            Request.Builder().url(server.url("")).build()
+        ).execute()
+
+        server.takeRequest().getHeader(name) shouldEqual value
```

Comment:
where does `value` get set?

Reply:
Each array from the collection below gets expanded and passed as the values to the constructor.